File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.py
class Solution:
    def numTrees(self, n: int) -> int:
        if n == 0 or n == 1:
            return 1

        catalan = 1
        for i in range(1, n + 1):
            catalan = (catalan * (4 * i - 2)) // (i + 1)
        
        return catalan

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.js
/**
 * @param {number} n
 * @return {number}
 */
var numTrees = function (n) {
    if (n === 0 || n === 1) return 1;

    let catalan = 1;
    for (let i = 1; i <= n; i++) {
        catalan = (catalan * (4 * i - 2)) / (i + 1);
    }
    
    return catalan;
};